extensions/refresh-catalog/src/main/java/org/eclipse/dataspaceconnector/mvd/FederatedCacheNodeResolver.java

```diff
@@ -40,21 +42,23 @@ public FederatedCacheNodeResolver(DidResolverRegistry resolver, Monitor monitor)
         this.monitor = monitor;
     }
 
-    public FederatedCacheNode toFederatedCacheNode(Participant participant) {
-        Result<DidDocument> didDocument = resolver.resolve(participant.getDid());
+    public Result<FederatedCacheNode> toFederatedCacheNode(Participant participant) {
+        String did = participant.getDid();
+        Result<DidDocument> didDocument = resolver.resolve(did);
```

Comment:
consider adding debug statement akin to @marcgs comment at https://github.com/agera-edc/RegistrationService/pull/19#discussion_r938528269